pydantic_ai_slim/pydantic_ai/_agent_graph.py

```diff
@@ -350,9 +355,10 @@ async def _make_request(
 
         model_settings, model_request_parameters = await self._prepare_request(ctx)
         model_request_parameters = ctx.deps.model.customize_request_parameters(model_request_parameters)
-        model_response = await ctx.deps.model.request(
-            ctx.state.message_history, model_settings, model_request_parameters
-        )
+        message_history = ctx.state.message_history
+        if ctx.deps.history_processors:
+            message_history = await _process_message_history(message_history, ctx.deps.history_processors)
```

Comment:
I think we can simplify this slightly here and above as `_process_message_history` just passes the history through if there are no processors

```suggestion
        message_history = await _process_message_history(ctx.state.message_history, ctx.deps.history_processors)
```